Cílem benchmarku je vytvořit jednoduchou referenční úroveň pro predikci inflace v ČR.

Použity jsou dva přístupy:
1.Naivní baseline model
2.ednoduchý lineární regresní model

Benchmarky jsou vyhodnoceny pouze na validační sadě. Testovací sada zůstává stranou pro pozdější finální vyhodnocení.

IMPORT KNIHOVEN


In [14]:
import pandas as pd
import numpy as np

from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, mean_absolute_error

NAČTENÍ DATASETŮ

In [15]:
train = pd.read_csv("data/train.csv", parse_dates=["date"], index_col="date")
valid = pd.read_csv("data/validation.csv", parse_dates=["date"], index_col="date")
test = pd.read_csv("data/test.csv", parse_dates=["date"], index_col="date")

print("Train:", train.shape)
print("Validation:", valid.shape)
print("Test:", test.shape)

Train: (75, 7)
Validation: (20, 7)
Test: (19, 7)


NAIVNÍ BASELNI

In [16]:
y_valid = valid["target"]
y_pred_baseline = valid["baseline"]

rmse_baseline = np.sqrt(mean_squared_error(y_valid, y_pred_baseline))
mae_baseline = mean_absolute_error(y_valid, y_pred_baseline)

print("Baseline RMSE:", rmse_baseline)
print("Baseline MAE:", mae_baseline)

Baseline RMSE: 0.7626717103769257
Baseline MAE: 0.5762882055624964


JEDNODUCHY ML BENCHMARK LINEÁRNÍ REGRESE

In [17]:
features = ["infl_de", "rate", "oil", "gdp_growth"]

X_train = train[features]
y_train = train["target"]

X_valid = valid[features]
y_valid = valid["target"]

model = LinearRegression()
model.fit(X_train, y_train)

y_pred_lr = model.predict(X_valid)

rmse_lr = np.sqrt(mean_squared_error(y_valid, y_pred_lr))
mae_lr = mean_absolute_error(y_valid, y_pred_lr)

print("Linear Regression RMSE:", rmse_lr)
print("Linear Regression MAE:", mae_lr)

Linear Regression RMSE: 2.239224577857834
Linear Regression MAE: 1.8280402314245645


POROVNÁNÍ VÝSLEDNŮ

In [18]:
results = pd.DataFrame({
    "model": ["Baseline", "Linear Regression"],
    "RMSE": [rmse_baseline, rmse_lr],
    "MAE": [mae_baseline, mae_lr]
})

results

,model,RMSE,MAE
0,Baseline,0.762672,0.576288
1,Linear Regression,2.239225,1.828040


## Vyhodnocení benchmarků

Naivní baseline model dosahuje lepších výsledků než lineární regresní model.

Baseline dosáhl RMSE = 0.762672, lineární regrese RMSE = 2.239225.

To naznačuje, že inflace má silnou setrvačnost a minulá hodnota inflace je velmi silným prediktorem budoucí hodnoty.

Lineární model využívající makroekonomické proměnné nedokázal tuto jednoduchou strategii překonat, což může být způsobeno omezeným množstvím dat a slabší predikční silou použitých proměnných.

Výsledek ukazuje, že baseline model představuje silnou referenční úroveň pro další modelování.

KOEFICINTY LINEÁRNÍHO MODELU

In [19]:
coef = pd.DataFrame({
    "feature": features,
    "coefficient": model.coef_
})

coef

,feature,coefficient
0,infl_de,1.329000
1,rate,0.611373
2,oil,0.014943
3,gdp_growth,-0.024785
